src/arch/svm/SpokeUtils.ts

```diff
@@ -81,27 +85,136 @@ export function findDepositBlock(
 }
 
 /**
- * Find the amount filled for a deposit at a particular block.
- * @param spokePool SpokePool contract instance.
- * @param relayData Deposit information that is used to complete a fill.
- * @param blockTag Block tag (numeric or "latest") to query at.
- * @returns The amount filled for the specified deposit at the requested block (or latest).
+ * Resolves the fill status of a deposit at a specific slot or at the current confirmed one.
+ *
+ * If no slot is provided, attempts to solve the fill status using the PDA. Otherwise, it is reconstructed from PDA events.
+ *
+ * @param programId - The spoke pool program ID.
+ * @param relayData - Deposit information used to locate the fill status.
+ * @param destinationChainId - Destination chain ID (must be an SVM chain).
+ * @param provider - SVM provider instance.
+ * @param svmEventsClient - SVM events client for querying events.
+ * @param atHeight - (Optional) Specific slot number to query. Defaults to the latest confirmed slot.
+ * @returns The fill status for the deposit at the specified or current slot.
  */
-export function relayFillStatus(
-  _spokePool: unknown,
-  _relayData: RelayData,
-  _blockTag?: number | "latest",
-  _destinationChainId?: number
+export async function relayFillStatus(
+  programId: Address,
+  relayData: RelayData,
+  destinationChainId: number,
+  provider: Provider,
+  svmEventsClient: SvmCpiEventsClient,
+  atHeight?: number
 ): Promise<FillStatus> {
-  throw new Error("relayFillStatus: not implemented");
+  assert(chainIsSvm(destinationChainId), "Destination chain must be an SVM chain");
+
+  // Get fill status PDA using relayData
+  const fillStatusPda = await getFillStatusPda(programId, relayData, destinationChainId);
+
+  // If no specific slot is requested, try fetching the current status from the PDA
+  if (atHeight === undefined) {
+    const fillStatusAccount = await fetchEncodedAccount(provider, fillStatusPda, { commitment: "confirmed" });
+
+    // If the PDA exists, return the stored fill status
+    if (fillStatusAccount.exists) {
+      const decodedAccountData = decodeFillStatusAccount(fillStatusAccount);
+      return decodedAccountData.data.status;
+    }
+    // If the PDA doesn't exist and the deadline hasn't passed yet, the deposit must be unfilled,
+    // since PDAs can't be closed before the fill deadline.
+    else if (getCurrentTime() < relayData.fillDeadline) {
```

Comment:
At the current implementation `getCurrentTime()` returns cached value from the latest `update()` call. This might be lower than the latest block time at the confirmed commitment when fill status account was fetched above, so there is some chance that this was filled and fill status account was closed just between the `update()` and `fetchEncodedAccount` calls. It would be safer to use the latest block time with the same confirmed commitment updating it immediately after the `fetchEncodedAccount` call.